ddtrace/contrib/tornado/decorators.py

```diff
@@ -24,8 +25,14 @@ def _finish_span(future):
         elif callable(getattr(future, 'exception', None)):
             # retrieve the exception from the Future object
             # that is executed in a different Thread
-            if future.exception():
-                span.set_exc_info(*sys.exc_info())
+            exc = future.exception()
+            if exc:
+                if PY3:
```

Comment:
Can you add a comment on why we need to do this.

Reply:
I added more conditions to this code with better documentation.